src/restruct_poc/beams.hpp

```diff
@@ -1,5 +1,175 @@
+    View_Nx4 qp_r;                  // State: rotation
+    View_Nx4 qp_r_prime;            // State: rotation derivative
+    View_Nx3 qp_omega;              // State: angular velocity
+    View_Nx3 qp_omega_dot;          // State: position/rotation
+    View_Nx3x4 qp_E;                // Quaternion derivative
+    View_Nx3x3 qp_eta_tilde;        //
+    View_Nx3x3 qp_omega_tilde;      //
+    View_Nx3x3 qp_omega_dot_tilde;  //
+    View_Nx3x3 qp_x0pupss;          //
+    View_Nx3x3 qp_M_tilde;          //
+    View_Nx3x3 qp_N_tilde;          //
+    View_Nx3 qp_eta;                //
+    View_Nx3x3 qp_rho;              //
+    View_Nx6 qp_strain;             // Strain
+    View_Nx6 qp_Fc;                 // Elastic force
+    View_Nx6 qp_Fd;                 // Elastic force
+    View_Nx6 qp_Fi;                 // Inertial force
+    View_Nx6 qp_Fg;                 // Gravity force
+    View_Nx6x6 qp_RR0;              // Global rotation
+    View_Nx6x6 qp_Muu;              // Mass in global frame
+    View_Nx6x6 qp_Cuu;              // Stiffness in global frame
+    View_Nx6x6 qp_Ouu;              // Linearization matrices
+    View_Nx6x6 qp_Puu;              // Linearization matrices
+    View_Nx6x6 qp_Quu;              // Linearization matrices
+    View_Nx6x6 qp_Guu;              // Linearization matrices
+    View_Nx6x6 qp_Kuu;              // Linearization matrices
+
+    View_NxN shape_interp;  // shape function matrix for interpolation [Nodes x QPs]
+    View_NxN shape_deriv;   // shape function matrix for derivative interp [Nodes x QPs]
+
+    // Scratch variables to be replaced later
+    View_Nx6x6 M_6x6;   //
+    View_Nx3x3 M8_3x3;  //
+    View_Nx3x3 M9_3x3;  //
+    View_Nx3 V2_3;      //
+    View_Nx3 V3_3;      //
+
+    Beams() {}  // Default constructor which doesn't initialize views
+
+    // Constructor which initializes views based on given sizes
+    Beams(
+        const int num_beams_, const int num_nodes_, const int num_qps_, const int max_elem_nodes_,
+        const int max_elem_qps_
+    )
+        : num_elems(num_beams_),
+          num_nodes(num_nodes_),
+          num_qps(num_qps_),
+          max_elem_nodes(max_elem_nodes_),
+          max_elem_qps(max_elem_qps_),
+          // Element Data
+          elem_indices("elem_indices", num_beams_),
+          node_state_indices("node_state_indices", num_nodes_),
+          gravity("gravity"),
+          // Node Data
+          node_x0("node_x0", num_nodes_),
+          node_u("node_u", num_nodes_),
+          node_u_dot("node_u_dot", num_nodes_),
+          node_u_ddot("node_u_ddot", num_nodes_),
+          node_FE("node_force_elastic", num_nodes_),
+          node_FG("node_force_gravity", num_nodes_),
+          node_FI("node_force_inertial", num_nodes_),
+          node_FX("node_force_external", num_nodes_),
+          node_residual("node_residual", num_nodes_ * 6),
+          node_iteration("node_iteration", num_nodes_ * 6, num_nodes_ * 6),
```

Comment:
These members need to be removed. They were originally going to be the fully assembled iteration matrix and residual vector for all beams, but the assembly uses the solver views directly.

Reply:
Good idea.  Will do